extras/R6ClassFun.R

```diff
@@ -0,0 +1,553 @@
+        makeModelDesignSettings(
+          targetId = exposureOfInterestIds[i],
+          outcomeId = outcomeOfInterestIds[j],
+          popSettings = plpPopulationSettings,
+          covarSettings = plpCovarSettings
+        )
+      )
+    )
+  }
+}
+
+plpModuleSpecifications <- plpModuleSettingsCreator$createModuleSpecifications(
+  modelDesignList = modelDesignList
+)
+
+
+
+
+# Create analysis specifications ---------------
+analysisSpecifications <- createEmptyAnalysisSpecificiations() |>
+  addSharedResources(cohortSharedResourcesSpecifications) |>
+  addSharedResources(ncoCohortSharedResourceSpecifications) |>
+  # WORKING
+  addModuleSpecifications(cgModuleSettings) |>
+  addModuleSpecifications(cdModuleSpecifications) |>
+  addModuleSpecifications(cmModuleSpecifications) |>
+  addModuleSpecifications(sccsModuleSpecifications) |>
+  addModuleSpecifications(plpModuleSpecifications)
+# NOT WORKING
+  #addModuleSpecifications(cModuleSpecifications) |>
+  # MOSTLY WORKING
+  #addModuleSpecifications(ciModuleSettings) |>
+
+# Cleanup any prior results -----------------
+outputFolder <- "D:/TEMP/StrategusR6Testing"
+unlink(outputFolder, recursive = T, force = T)
+dir.create(outputFolder, recursive = T)
+
+# Execute -------------------
+ParallelLogger::saveSettingsToJson(analysisSpecifications, file.path(outputFolder, "analysisSettings.json"))
+workFolder <- file.path(outputFolder, "work_folder")
+resultsFolder <- file.path(outputFolder, "results_folder")
+
+executionSettings <- Strategus::createCdmExecutionSettings(
+  workDatabaseSchema = "main",
+  cdmDatabaseSchema = "main",
+  cohortTableNames = CohortGenerator::getCohortTableNames(cohortTable = "strategus_test"),
+  workFolder = workFolder,
+  resultsFolder = resultsFolder,
+  minCellCount = 5
+)
+
+connectionDetails <- Eunomia::getEunomiaConnectionDetails(
+  databaseFile = file.path(outputFolder, "Eunomia.sqlite"),
+  overwrite = TRUE
+)
+#debugonce(Strategus::execute)
+Strategus::execute(
+  analysisSpecifications = analysisSpecifications,
+  executionSettings = executionSettings,
+  connectionDetails = connectionDetails
+)
+
+# # # DEBUG CD
+# cdModule <- CohortDiagnosticsModule$new()
+# debugonce(cdModule$execute)
+# executionSettings$databaseId = "Eunomia"
+# cdModule$execute(
+#   analysisSpecifications = analysisSpecifications,
+#   executionSettings = executionSettings,
+#   connectionDetails = connectionDetails
+# )
+
+# Create empty results database -------------------------
+library(RSQLite)
+mydb <- dbConnect(RSQLite::SQLite(), file.path(outputFolder, "results.sqlite"))
+dbDisconnect(mydb)
+
+resultsConnectionDetails <- DatabaseConnector::createConnectionDetails(
+  dbms = "sqlite",
+  server = file.path(outputFolder, "results.sqlite")
+)
+
+# Create results schema tables -------------------------
+# NOTE: resultsExecutionSettings uses a diff. folder to hold
+# any/all results schema creation results
+resultsExecutionSettings <- Strategus::createResultsExecutionSettings(
```

Comment:
Noting feedback from @jreps that it would be helpful & more descriptive if the settings were named after the intended function. In this context, `createResultDataModelSettings` vs. `createResultsExecutionSettings` and the same for upload.

Reply:
Refactored things in https://github.com/OHDSI/Strategus/pull/145/commits/15ade7869991c7eebdcc273aead762724823d85f to add new settings functions for the results data model creation & upload. Noting here that I'm now exposing the `purgeSiteDataBeforeUploading` parameter from RMM but we seem to lack a way to pass in the database identifier file to many of the modules to make use of this setting. Tagging @azimov for review of this specific item.